src/ddb/ddb_commands.c

```diff
@@ -619,3 +620,121 @@ ddb_run_smd_sync(struct ddb_ctx *ctx)
 	ddb_printf(ctx, "Done: "DF_RC"\n", DP_RC(rc));
 	return rc;
 }
+
+struct dump_vea_cb_args {
+	struct ddb_ctx *dva_ctx;
+	uint32_t	dva_count;
+};
+
+static int
+dump_vea_cb(void *cb_arg, struct vea_free_extent *vfe)
+{
+	struct dump_vea_cb_args *args = cb_arg;
+
+	ddb_printf(args->dva_ctx, "[Region %d] offset: %lu, block count: %d, age: %d\n",
+		   args->dva_count,
+		   vfe->vfe_blk_off,
+		   vfe->vfe_blk_cnt,
+		   vfe->vfe_age);
+
+	args->dva_count++;
+	return 0;
+}
+
+int
+ddb_run_dump_vea(struct ddb_ctx *ctx)
+{
+	struct dump_vea_cb_args args = {.dva_ctx = ctx, .dva_count = 0};
+	int			rc;
+
+	rc = dv_enumerate_vea(ctx->dc_poh, dump_vea_cb, &args);
+
+	ddb_printf(ctx, "Total Free Regions: %d\n", args.dva_count);
+
+	return rc;
+}
+
+static int
+parse_uint32_t(char *str)
+{
+	uint32_t result = atoi(str);
+	char	 verify_str[32];
+
+	snprintf(verify_str, ARRAY_SIZE(verify_str), "%d", result);
+
+	if (strcmp(str, verify_str) == 0)
+		return result;
+
+	return -DER_INVAL;
+
+}
+
+struct update_vea_verify_region_cb_args {
+	struct ddb_ctx		*ctx;
+	struct vea_free_extent	 potential_extent;
+};
+
+/**
+ *
+ * @param n - new extent to insert or update
+ * @param e - existing extent
+ * @return
+ */
+static bool
+vfe_overlap(struct vea_free_extent *n, struct vea_free_extent *e)
+{
+	uint64_t a_lo = n->vfe_blk_off;
+	uint64_t a_hi = n->vfe_blk_off + n->vfe_blk_cnt - 1;
+	uint64_t b_lo = e->vfe_blk_off;
+	uint64_t b_hi = e->vfe_blk_off + e->vfe_blk_cnt - 1;
+
+	return !(a_hi < b_lo || a_lo > b_hi);
+}
+
+static int
+update_vea_verify_region_cb(void *cb_arg, struct vea_free_extent *vfe)
+{
+	struct update_vea_verify_region_cb_args *args = cb_arg;
+
```

Comment:
Check if the 'vfe_blk_off' of both extents are > 0 before the overlapping check.

Reply:
vfe comes from what's already in the table, so the offset should never be 0, right? There is a check to make sure the args->potential_extent offset is not 0 on line 719.